lib/streaming.js

```diff
@@ -110,11 +119,17 @@ StreamProxy.prototype._onFinish = function() {
 StreamProxy.prototype.setStream = function(apiCall, argument) {
   var stream = apiCall(argument, this._callback);
   this.stream = stream;
-  stream.on('error', this.emit.bind(this, 'error'));
+  (['error', 'metadata', 'status']).forEach(function(event) {
+    stream.on(event, this.emit.bind(this, event));
+  }.bind(this));
+  // We also want to supply the status data as 'response'.
+  stream.on('status', this.emit.bind(this, 'response'));
   if (this.type !== StreamType.CLIENT_STREAMING) {
     stream.on('data', this.emit.bind(this, 'data'));
     // Pushing null causes an ending process of the readable stream.
     stream.on('end', this.push.bind(this, null));
+    // This is required in case there's no 'data' handler exists.
```

Comment:
"in case there's no 'data' handler exists" --> "in case no 'data' handler exists"